### PR TITLE
[BACK-3044] Add LastReviewed to Patient Record

### PR DESCRIPTION
**Pull request by Roukoswarf**

### CODE REVIEW

File: spec/clinic.v1.yaml [hunk 1]
```diff
@@ -2962,6 +2996,10 @@ components:
           readOnly: true
         summary:
           $ref: '#/components/schemas/PatientSummary'
+        lastReviewed:
```
Comment (toddkazakov) [inline spec/clinic.v1.yaml:2999]: This two new fields ought to be an array

File: patients/patients.go [hunk 2]
```diff
@@ -44,6 +44,7 @@ type Service interface {
 	List(ctx context.Context, filter *Filter, pagination store.Pagination, sort []*store.Sort) (*ListResult, error)
 	Create(ctx context.Context, patient Patient) (*Patient, error)
 	Update(ctx context.Context, update PatientUpdate) (*Patient, error)
+	UpdateLastReviewed(ctx context.Context, update PatientUpdate) (*Patient, error)
```
Comment (toddkazakov) [inline patients/patients.go:47]: Updates of the patient object should use the `update` method of the service. It even has the exact same signature as this one.

File: patients/repo.go [hunk 3]
```diff
@@ -334,6 +334,35 @@ func (r *repository) UpdateEmail(ctx context.Context, userId string, email *stri
...
+	opts := options.FindOneAndUpdate().SetReturnDocument(1) // return document AFTER edit
+	selector := bson.M{
+		"clinicId": clinicObjId,
+		"userId":   patientUpdate.UserId,
+	}
+
+	patient := patientUpdate.Patient
+	patient.UpdatedTime = time.Now()
+
+	update := bson.M{
```
Comment (toddkazakov) [inline patients/repo.go:348]: This will reset fields without `omitempty` to their defaults (e.g. `requireUniqueMrn`).

File: api/patients.go [hunk 4]
```diff
@@ -443,3 +443,103 @@ func (h *Handler) FindPatients(ec echo.Context, params FindPatientsParams) error
...
+			LastReviewed: &patients.LastReviewed{
+				ClinicianId: clinicianId,
+				Time:        time.Now().UTC().Truncate(time.Millisecond),
+			},
+		},
+		ClinicId: clinicId,
+		UserId:   patientId,
+	}
+
+	if patient.LastReviewed != nil {
```
Comment (toddkazakov) [inline api/patients.go:481]: This logic should be in the patients service

File: api/patients.go [hunk 5]
```diff
@@ -443,3 +443,103 @@ func (h *Handler) FindPatients(ec echo.Context, params FindPatientsParams) error
...
+	}
+
+	clinicianId := authData.SubjectId
+
+	patient, err := h.patients.Get(ctx, clinicId, patientId)
+	if err != nil {
+		return err
+	}
+
+	if patient.PreviousLastReviewed == nil {
```
Comment (toddkazakov) [inline api/patients.go:516]: This logic should be in the patients service

**Review by toddkazakov — CHANGES_REQUESTED** [review]

File: spec/clinic.v1.yaml (hunk 1 above)
Comment (Roukoswarf) [reply spec/clinic.v1.yaml:2999]: done

File: patients/repo.go (hunk 3 above)
Comment (Roukoswarf) [reply patients/repo.go:348]: moved away from patientUpdate

File: patients/patients.go (hunk 2 above)
Comment (Roukoswarf) [reply patients/patients.go:47]: moved away from full patient update, just updating the single review field now.

**Review by lostlevels — APPROVED** [review]

File: patients/repo.go [hunk 6]
```diff
@@ -334,6 +334,72 @@ func (r *repository) UpdateEmail(ctx context.Context, userId string, email *stri
...
+		"$push": bson.M{
+			"reviews": bson.M{
+				"$each":     []Review{review},
+				"$position": 0,
+				"$slice":    2,
+			},
+		},
+	}
+
+	opts := options.FindOneAndUpdate().SetReturnDocument(1) // return document AFTER edit
```
[the remark this answers is not in the document]
Comment (Roukoswarf) [reply patients/repo.go:354]: didnt see the constant, that also removes the clarifying comment, nice, thank you.

File: patients/repo.go [hunk 7]
```diff
@@ -334,6 +334,72 @@ func (r *repository) UpdateEmail(ctx context.Context, userId string, email *stri
...
+	clinicObjId, _ := primitive.ObjectIDFromHex(clinicId)
+	selector := bson.M{
+		"clinicId":              clinicObjId,
+		"userId":                userId,
+		"reviews.0.clinicianId": clinicianId,
+	}
+
+	update := bson.M{"$pop": bson.M{"reviews": -1}}
+
+	opts := options.FindOneAndUpdate().SetReturnDocument(1) // return document AFTER edit
```
[the remark this answers is not in the document]
Comment (Roukoswarf) [reply patients/repo.go:377]: didnt see the constant, that also removes the clarifying comment, nice, thank you.

**Pull request merged**